### PR TITLE
Add a test for paginated search results

### PR DESCRIPTION
**Pull request by richvdh**
This tests the fix added at matrix-org/synapse#447

### CODE REVIEW

**Comment by richvdh** [comment]
... and it fails because the fix isn't on develop yet.

File: tests/43search.pl [hunk 1]
```diff
@@ -53,3 +54,109 @@
...
+            assert_json_keys( $body->{search_categories}, qw ( room_events ) );
+            my $room_events = $body->{search_categories}{room_events};
+
+            assert_json_keys( $room_events, qw( count results next_batch ) );
+
+            $room_events->{count} == 20 or die "Expected count == 20";
+
+            my $results = $room_events->{results};
+            scalar @$results == 10 or die "Expected 10 search results";
+            $results->[0]{result}{event_id} eq $event_ids[19]
```
Comment (leonerd) [inline tests/43search.pl]: These can be done with

    assert_eq( $results->[0]{result}{event_id}, $event_ids[19],
       'first result event_id' );

which prints the expected/actual values on failure

File: tests/43search.pl [hunk 2]
```diff
@@ -53,3 +54,109 @@
...
+                }
+            }
+        };
+
+        repeat( sub {
+            my $msgnum = $_[0];
+            matrix_send_room_text_message( $user, $room_id,
+                                           body => "Message number $msgnum" )
+                ->on_done( sub { ($event_ids[$msgnum]) = @_ } )
+        }, foreach => [ 0 .. 19 ] ) -> then ( sub {
```
Comment (leonerd) [inline tests/43search.pl]: Whitespace:

    ( $event_ids[$msgnum] ) = @_

and

    )->then( sub {

**Comment by leonerd** [comment]
Overall: the test appears to be expecting that search results come in batches of 10 each, but the search parameters didn't request exactly 10. This feels like it's a fragile assumption of implementation. Is there perhaps a limit parameter that can be specified?

File: tests/43search.pl (hunk 1 above)
Comment (richvdh) [inline tests/43search.pl]: that's new!

Done.

File: tests/43search.pl (hunk 2 above)
Comment (richvdh) [inline tests/43search.pl]: Done.

**Comment by richvdh** [comment]
good point about the implicit limit. Now made explicit.

**Comment by leonerd** [comment]
LGTM

**Comment by richvdh** [comment]
ta

**Pull request merged**